src/middleware/persist.ts

```diff
@@ -5,19 +5,15 @@ import {
   StoreMutatorIdentifier,
 } from '../vanilla'
 
-type DeepPartial<T> = {
-  [P in keyof T]?: DeepPartial<T[P]>
-}
-
 export type StateStorage = {
   getItem: (name: string) => string | null | Promise<string | null>
   setItem: (name: string, value: string) => void | Promise<void>
   removeItem: (name: string) => void | Promise<void>
 }
 
-type StorageValue<S> = { state: DeepPartial<S>; version?: number }
+type StorageValue<S> = { state: S; version?: number }
 
-export type PersistOptions<S, PersistedState = Partial<S>> = {
+export type PersistOptions<S, PersistedState = S> = {
```

Comment:
Here again the types are lying, the default persisted state when no partialize is provided is the whole state ie S and not Partial<S>.